ssl/record/methods/tls_common.c

```diff
@@ -1238,6 +1238,8 @@ static void tls_int_free(OSSL_RECORD_LAYER *rl)
     if (rl->version == SSL3_VERSION)
         OPENSSL_cleanse(rl->mac_secret, sizeof(rl->mac_secret));
 
+    SSL3_RECORD_release(rl->rrec, rl->num_recs);
```

Comment:
This should be `SSL_MAX_PIPELINES` not `rl->num_recs`. The number of active records at any one time may vary, so the current value for `rl->num_recs` may be less than the maximum value it has ever been. By using `SSL_MAX_PIPELINES` instead we release all records regardless of whether they are currently in use or not.

Reply:
fixed